src/NuGet.Core/NuGet.Packaging/Signing/ChainBuilding/X509ChainBuildPolicyFactory.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    internal static class X509ChainBuildPolicyFactory
+    {
+        // These fields are non-private only to facilitate testing.
+        internal const string EnvironmentVariableName = "NUGET_EXPERIMENTAL_CHAIN_BUILD_RETRY_POLICY";
+        internal const char ValueDelimiter = ',';
+
+        internal static IX509ChainBuildPolicy Create(IEnvironmentVariableReader reader)
+        {
+            if (reader is null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                string value = reader.GetEnvironmentVariable(EnvironmentVariableName);
+
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    return DefaultX509ChainBuildPolicy.Instance;
+                }
+
+                string[] parts = value.Split(ValueDelimiter);
```

Comment:
nit: Since you're expecting a max of 2 entries, you might as well limit the `Split()` as well

```C#
private readonly static char[] ValueDelimiter = new[] { ',' };
...
string[] parts = @value.Split(ValueDelimiter, 2);
```

Reply:
No, I don't want to do that.  If the value contains more than one delimiter, then it's not being used correctly.